File: sdks/php/src/Wallets/Requests/CreateWalletDto.php
<?php

namespace NovaBilling\Wallets\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class CreateWalletDto extends JsonSerializableType
{
    /**
     * @var string $customerId
     */
    #[JsonProperty('customerId')]
    public string $customerId;

    /**
     * @var ?string $name
     */
    #[JsonProperty('name')]
    public ?string $name;

    /**
     * @var string $currency
     */
    #[JsonProperty('currency')]
    public string $currency;

    /**
     * @var ?float $rateAmount 1 credit = rateAmount in currency
     */
    #[JsonProperty('rateAmount')]
    public ?float $rateAmount;

    /**
     * @var ?float $paidCredits Paid credits (purchase)
     */
    #[JsonProperty('paidCredits')]
    public ?float $paidCredits;

    /**
     * @var ?float $grantedCredits Free credits (grant)
     */
    #[JsonProperty('grantedCredits')]
    public ?float $grantedCredits;

    /**
     * @var ?string $expirationAt Expiration date (ISO 8601)
     */
    #[JsonProperty('expirationAt')]
    public ?string $expirationAt;

    /**
     * @var ?array<string, mixed> $metadata
     */
    #[JsonProperty('metadata'), ArrayType(['string' => 'mixed'])]
    public ?array $metadata;

    /**
     * @var ?string $createdAt Backdate createdAt (ISO 8601). For data imports.
     */
    #[JsonProperty('createdAt')]
    public ?string $createdAt;

    /**
     * @param array{
     *   customerId: string,
     *   currency: string,
     *   name?: ?string,
     *   rateAmount?: ?float,
     *   paidCredits?: ?float,
     *   grantedCredits?: ?float,
     *   expirationAt?: ?string,
     *   metadata?: ?array<string, mixed>,
     *   createdAt?: ?string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->customerId = $values['customerId'];
        $this->name = $values['name'] ?? null;
        $this->currency = $values['currency'];
        $this->rateAmount = $values['rateAmount'] ?? null;
        $this->paidCredits = $values['paidCredits'] ?? null;
        $this->grantedCredits = $values['grantedCredits'] ?? null;
        $this->expirationAt = $values['expirationAt'] ?? null;
        $this->metadata = $values['metadata'] ?? null;
        $this->createdAt = $values['createdAt'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/Wallets/Requests/CreateWalletDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record CreateWalletDto
{
    [JsonPropertyName("customerId")]
    public required string CustomerId { get; set; }

    [JsonPropertyName("name")]
    public string? Name { get; set; }

    [JsonPropertyName("currency")]
    public required string Currency { get; set; }

    /// <summary>
    /// 1 credit = rateAmount in currency
    /// </summary>
    [JsonPropertyName("rateAmount")]
    public double? RateAmount { get; set; }

    /// <summary>
    /// Paid credits (purchase)
    /// </summary>
    [JsonPropertyName("paidCredits")]
    public double? PaidCredits { get; set; }

    /// <summary>
    /// Free credits (grant)
    /// </summary>
    [JsonPropertyName("grantedCredits")]
    public double? GrantedCredits { get; set; }

    /// <summary>
    /// Expiration date (ISO 8601)
    /// </summary>
    [JsonPropertyName("expirationAt")]
    public string? ExpirationAt { get; set; }

    [JsonPropertyName("metadata")]
    public Dictionary<string, object?>? Metadata { get; set; }

    /// <summary>
    /// Backdate createdAt (ISO 8601). For data imports.
    /// </summary>
    [JsonPropertyName("createdAt")]
    public string? CreatedAt { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
